### PR TITLE
mounter(ticdc): ignore checksum v2

### LINKED ISSUE #11524: ignore checksum if the version is not v1
For release-8.1 and release-7.5, only verify checksum v1.

In the upgrade scenario, the upstream TiDB is upgraded first, then may output checksum v2, in the such case, downstream TiCDC cannot verify it, and makes the changefeed error, so just ignore the error

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: close #11524

### What is changed and how it works?

* ignore the checksum if not v1 (version != 0)

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/entry/mounter.go [hunk 1]
```diff
@@ -434,6 +434,13 @@ func (m *mounter) verifyChecksum(
 		decoder = m.decoder
 	}
 
+	version := decoder.ChecksumVersion()
+	if version != 0 {
+		log.Info("checksum is not v1, this happens since upstream TiDB is in a higher version, ignore it",
```
Comment (asddongmen) [inline cdc/entry/mounter.go:439]: Would this log flooding?

Comment (3AceShowHand) [reply cdc/entry/mounter.go:439]: this only happens in the upgrade scenario, it should not take too long, so not too match logs.

**Review by sdojjy — APPROVED** [review]

File: cdc/entry/mounter.go (hunk 1 above)
Comment (asddongmen) [reply cdc/entry/mounter.go:439]: Rolling upgrade would take more than 10 minutes if there are a lot of tables.

**Review by asddongmen — APPROVED** [review]

**Pull request merged**